src/PackagePublisher/Program.cs

```diff
@@ -24,7 +24,9 @@ public static class Program
 
         // Do NOT log the key
         private static string _apiKey = null;
-        private static int _maxRetryCount = 10;
+        private static int _maxRetryCount = 5;
+        private static int _packagePushTimeoutInSeconds = 90;
+        private static int _semaphoreTimeoutInSeconds = _maxRetryCount * _packagePushTimeoutInSeconds;
```

Comment:
Earlier the semaphore timeout was 3 mins (or 180 seconds) where as the worst case of a push would have been 10 * 30 seconds resulting in semaphore entry failure which resulted number of parallel pushes to be greater than 4 and also semaphore release to be called incorrectly.